demos/crossroad_camera_demo/main.cpp

```diff
@@ -161,23 +161,19 @@ struct PersonDetection : BaseDetection{
     }
 
     PersonDetection() : BaseDetection(FLAGS_m, "Person Detection"), maxProposalCount(0), objectSize(0) {}
-    CNNNetwork read() override {
+    CNNNetwork read(const Core& ie) override {
         slog::info << "Loading network files for PersonDetection" << slog::endl;
-        CNNNetReader netReader;
         /** Read network model **/
-        netReader.ReadNetwork(FLAGS_m);
-        /** Extract model name and load it's weights **/
-        std::string binFileName = fileNameNoExt(FLAGS_m) + ".bin";
-        netReader.ReadWeights(binFileName);
+        auto network = ie.ReadNetwork(FLAGS_m);
```

Comment:
I think you should specify the path to the weights explicitly here. The documentation says that if the second argument is omitted and a file with the automatically-derived name isn't found, the function will not load the weights at all, which is not what we want. We want it to fail with a file-not-found error.

Reply:
@IRDonch if IR contains some weighs but you don't provide **\*.bin** file, the network won't be read and you will get error that some layer cannot set weights.

Reply:
I have experimented with this, and the errors you get when you specify the weights path and when you don't are quite different.

When you don't, `ReadNetwork` just crashes.

When you do, it throws an exception:

```
terminate called after throwing an instance of 'InferenceEngine::details::InferenceEngineException'
  what():  filesize for: test.bin - -1<0. Please, check weights file existence.
```

The latter is certainly preferable to the former, as it hints as to what went wrong.

Reply:
I created a PR to OpenVINO which add more informative error message if model requires weights, but bin file wasn't found.

Reply:
All right then, sounds good to me.